committeeoversightapp/management/commands/import_data.py

```diff
@@ -115,98 +109,165 @@ def add_house_hearings(self):
+
+        if organization_exact.count() == 1:
+            return organization_exact[0]
         else:
-            return ValueError
+            if organization_contains.count() == 1:
+                return organization_contains[0]
+            elif organization_contains.count() == 0:
+                organization_other_name = Organization.objects.filter(other_names__name__icontains=committee_name, parent=parent, classification=classification)
+                try:
+                    return organization_other_name[0]
+                except IndexError:
+                    new_committee, _ = Committee.objects.get_or_create(lugar_id=committee_key, lugar_name=committee_name)
+                    self.bad_rows.append("Not recognized: " + committee_key + ", " + committee_name)
+            else:
+                self.bad_rows.append("Multiple possible committees for " + committee_key + ": " + committee_name)
 
-    def new_committee(self, committee_key, event, row_count):
+    def new_event_participant(self, committee_key, event):
         try:
             name = Committee.objects.get(lugar_id=committee_key).organization.name
             organization = Committee.objects.get(lugar_id=committee_key).organization
             entity_type = "organization"
             committee, created = EventParticipant.objects.get_or_create(name=name, event=event, organization=organization, entity_type=entity_type)
 
         except (ObjectDoesNotExist, AttributeError, ValueError) as e:
-            self.bad_rows.append("Row " + str(row_count) + ": Bad committee key " + committee_key)
+            self.bad_rows.append("Row " + str(self.row_count) + ": Bad committee key " + committee_key)
+
+    def new_category(self, event, category):
+        try:
+            hearing_category, _ = HearingCategory.objects.get_or_create(event=event, category_id=category)
+
+        except IntegrityError:
+            self.bad_rows.append("Row " + str(self.row_count) + ": Unrecognized category " + category + " on " + event.name)
+
+    def new_source(self, event, note, url):
+        source, _ = EventSource.objects.get_or_create(event=event, note=note, url=url)
+
+    def smart_match(self, matched, name, committees_filtered, start_date, category, source, classification):
+        # try to match by date and committees
+        print("Attempting to match " + name)
+        try:
+            committee_qs = Committee.objects.filter(lugar_id__in=committees_filtered)
+            committee_set = set(committee.organization.name for committee in committee_qs)
+            matched_events = Event.objects.filter(participants__name__in=committee_set, start_date=start_date).distinct()
+        except ValueError:
+            matched_events = []
+            self.bad_rows.append("Row " + str(self.row_count) + ": Bad committee value in " + str(committees_filtered))
```

Comment:
this is about where this method should end. the rest of the code in this current method is not trying to match, but updating the results of a match.

Reply:
Thanks!